### PR TITLE
feat(keystone): DOMA-10930 added "GraphiQL" instead of the outdated "graphql-playground"

### PR DESCRIPTION
**Pull request by Alllex202**
- [x] use `nunjucks`
- [x] ability to change path to api
- [x] lock library versions
- [x] disable old playground
- [x] add static files 

## Preview

https://github.com/user-attachments/assets/7102f5e8-cb1f-4e8f-a25f-8c357d093382

### CODE REVIEW

File: packages/keystone/graphiql/graphiql.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Playground - __REPLACE_TO_YOUR_SERVER_URL__</title>
```
Comment (sitozzz) [inline packages/keystone/graphiql/graphiql.html:6]:
```suggestion
    <title>Playground</title>
```

Comment (sitozzz) [reply packages/keystone/graphiql/graphiql.html:6]: Or even condo playground)

File: packages/keystone/graphiql/graphiql.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <link rel="stylesheet" href="https://unpkg.com/graphiql/graphiql.min.css" />
+</head>
+<body style="margin: 0;">
+<div id="graphiql" style="height: 100vh;"></div>
+<script crossorigin src="https://unpkg.com/react/umd/react.production.min.js"></script>
+<script crossorigin src="https://unpkg.com/react-dom/umd/react-dom.production.min.js"></script>
+<script crossorigin src="https://unpkg.com/graphiql/graphiql.min.js"></script>
+<script>
+    const apiPath = '__REPLACE_TO_YOUR_API_PATH__'
```
Comment (sitozzz) [inline packages/keystone/graphiql/graphiql.html:15]: That's not looking good)

Comment (sitozzz) [reply packages/keystone/graphiql/graphiql.html:15]: Oh yeah, I found the reason below

File: packages/keystone/graphiql/graphiql.html (hunk 1 above)
Comment (Alllex202) [reply packages/keystone/graphiql/graphiql.html:6]: The point here is to be able to differentiate between sandbox tabs of different applications.
This is exactly how it works for us now.
However, we can forget about this moment)

File: packages/keystone/graphiql/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        const html = fileBuffer.toString()
+        return html
+            .replace('__REPLACE_TO_YOUR_API_PATH__', apiPath)
+            .replace('__REPLACE_TO_YOUR_SERVER_URL__', SERVER_URL)
+    }
+
+    async prepareMiddleware () {
+        // nosemrep: javascript.express.security.audit.express-check-csurf-middleware-usage.express-check-csurf-middleware-usage
+        const app = express()
+        app.get(this._graphiqlPath, (req, res) => {
```
Comment (pahaz) [inline packages/keystone/graphiql/index.js:31]: may be we need to move it to another URL?

File: packages/keystone/graphiql/graphiql.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const graphQLFetcher = async (graphQLParams) =>
+        fetch(apiPath, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            credentials: 'omit',
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply packages/keystone/graphiql/graphiql.html:22]: The behavior matches what is in our current playground.

Authentication cookies are not taken from the platform. 
All requests are made with the auth token in the header

File: packages/keystone/graphiql/index.js (hunk 3 above)
Comment (Alllex202) [reply packages/keystone/graphiql/index.js:31]: We can try to leave the old playground, but move it to another URL. 
And leave the new playground on the current URL, so as not to update the links in all the documentation.

The old playground is almost impossible to use due to overlapping popups when writing queries, but there you can change URL to API

Comment (Alllex202) [reply packages/keystone/graphiql/index.js:30]: moved the new playground to another URL

**Review by AleX83Xpert — APPROVED** [review]

File: packages/keystone/KSv5v6/v5/prepareKeystone.js [hunk 5]
```diff
@@ -185,6 +188,7 @@ function prepareKeystone ({ onConnect, extendKeystoneConfig, extendExpressApp, s
             new IpBlackListMiddleware(),
             new KeystoneTracingApp(),
             ...((apps) ? apps() : []),
+            ...(IS_ENABLE_DANGEROUS_GRAPHQL_PLAYGROUND ? [new GraphiqlApp({ graphiqlPath: '/admin/graphiql' })] : []),
```
Comment (sitozzz) [inline packages/keystone/KSv5v6/v5/prepareKeystone.js:191]: What about disable existing one via passing `IS_ENABLE_DANGEROUS_GRAPHQL_PLAYGROUND` to `playground` property a little lower in the code?

File: packages/keystone/graphiql/graphiql.html (hunk 2 above)
Comment (sitozzz) [reply packages/keystone/graphiql/graphiql.html]: Now looks good at this point

File: packages/keystone/KSv5v6/v5/prepareKeystone.js (hunk 5 above)
Comment (Alllex202) [reply packages/keystone/KSv5v6/v5/prepareKeystone.js:191]: for now I decided not to remove the old version - now there are links from different documentation and also the ability to change the path to the API

Comment (Alllex202) [reply packages/keystone/KSv5v6/v5/prepareKeystone.js]: disabled old playground

File: packages/keystone/KSv5v6/v5/prepareKeystone.js [hunk 6]
```diff
@@ -184,12 +187,13 @@ function prepareKeystone ({ onConnect, extendKeystoneConfig, extendExpressApp, s
...
             new KeystoneTracingApp(),
             ...((apps) ? apps() : []),
+            ...(IS_ENABLE_DANGEROUS_GRAPHQL_PLAYGROUND ? [new GraphiqlApp()] : []),
             new GraphQLApp({
                 apollo: {
                     formatError: safeApolloErrorFormatter,
                     debug: IS_ENABLE_APOLLO_DEBUG,
                     introspection: IS_ENABLE_DANGEROUS_GRAPHQL_PLAYGROUND,
-                    playground: IS_ENABLE_DANGEROUS_GRAPHQL_PLAYGROUND,
+                    playground: false,
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply packages/keystone/KSv5v6/v5/prepareKeystone.js:203]: I don't think it's critical :)

**Review by sitozzz — APPROVED** [review]
Let's go

**Pull request merged**